pod/annotations.go

```diff
@@ -159,6 +159,14 @@ const (
 	// service configuration
 
 	AnnotationKeyServicePrefix = "service.netflix.com"
+
+	// sidecar configuration
+
+	AnnotationKeyPrefixSidecars                      = "titusParameter.agent.sidecars"
+	AnnotationKeySuffixSidecarsChannelOverride       = "channel-override"
+	AnnotationKeySuffixSidecarsChannelOverrideReason = "channel-override-reason"
```

Comment:
Whoa, we have an override-reason?

Reply:
Yes, the user can define a Starlark function that overrides the channel. This function returns the channel override, and also a reason (which ends up as an annotation on the pod).

Reply:
Aha this explains the "channel-reason" conversation we had earlier on Slack, @solarkennedy 😎 I think with this in mind, we won't need to capture this in the Job itself -- because the only source of this information is the Webhook, and thus this is an output attached to the Task rather than the Job.

Reply:
Yea but is just logging it good enough?

Reply:
We include the reason in the annotation so the app owners have visibility, since the channel used won't match the one they specified because of the override. [NCE-60](https://jira.netflix.net/browse/NCE-60) has a bit more context.

Reply:
Ok. I think just logging is probably an OK start for this.

Setting an annotation on this pod in this manner will not be visible to a user.